payu/models/access.py

```diff
@@ -89,61 +100,68 @@ def setup(self):
-                # interested in:
-                #   1. start date of run
-                #   2. total runtime of all previous runs.
+                # Get timing information for the new run.
                 if model.prior_restart_path and not self.expt.repeat_run:
-
-                    prior_cpl_fpath = os.path.join(model.prior_restart_path,
-                                                   cpl_fname)
-
-                    # With later versions this file exists in the prior restart
-                    # path, but this was not always the case, so check, and if
-                    # not there use prior output path
-                    if not os.path.exists(prior_cpl_fpath):
-                        print('payu: warning: {0} missing from prior restart '
-                              'path; checking prior output.'.format(cpl_fname),
-                              file=sys.stderr)
-                        if not os.path.isdir(model.prior_output_path):
-                            print('payu: error: No prior output path; '
-                                  'aborting run.')
-                            sys.exit(errno.ENOENT)
-
-                        prior_cpl_fpath = os.path.join(model.prior_output_path,
-                                                       cpl_fname)
+                    # Read the start date from the restart date namelist.
+                    start_date_fpath = os.path.join(
+                        model.prior_restart_path,
+                        model.start_date_nml_name
+                    )
 
                     try:
-                        prior_cpl_nml = f90nml.read(prior_cpl_fpath)
-                    except IOError as exc:
-                        if exc.errno == errno.ENOENT:
-                            print('payu: error: {0} does not exist; aborting.'
-                                  ''.format(prior_cpl_fpath), file=sys.stderr)
-                            sys.exit(exc.errno)
-                        else:
-                            raise
-
-                    cpl_nml_grp = prior_cpl_nml[cpl_group]
-
-                    # The total time in seconds since the beginning of
-                    # the experiment.
-                    total_runtime = int(cpl_nml_grp[runtime0_key] +
-                                        cpl_nml_grp['runtime'])
-                    run_start_date = cal.date_plus_seconds(init_date,
-                                                           total_runtime,
-                                                           caltype)
+                        start_date_nml = f90nml.read(start_date_fpath)[
+                            model.cpl_group]
+                    except FileNotFoundError:
+                        print(
+                            "Missing restart date file for model "
+                            f"{model.model_type}",
+                            file=sys.stderr
+                        )
```

Comment:
Will this report the path that is supposed to exist? I think it is always good to tell the user what the expected behaviour was, which in this case that there be a file of a certain name in a particular location.

Reply:
In the text for the FileNoteFoundError if will give the expected path